src/sentry/monitors/endpoints/base_monitor_details.py

```diff
@@ -112,32 +112,34 @@ def update_monitor(
         if "project" in result and result["project"].id != monitor.project_id:
             raise ParameterValidationError("existing monitors may not be moved between projects")
 
+        old_slug = monitor.slug
+        old_status = monitor.status
+        if params:
+            monitor.update(**params)
+            self.create_audit_entry(
+                request=request,
+                organization=project.organization,
+                target_object=monitor.id,
+                event=audit_log.get_event_id("MONITOR_EDIT"),
+                data=monitor.get_audit_log_data(),
+            )
+
         # Update monitor slug
         if "slug" in result:
-            quotas.backend.update_monitor_slug(monitor.slug, params["slug"], monitor.project_id)
+            quotas.backend.update_monitor_slug(old_slug, params["slug"], monitor.project_id)
 
         # Attempt to assign a monitor seat
-        if params["status"] == ObjectStatus.ACTIVE:
+        if params["status"] == ObjectStatus.ACTIVE and old_status != ObjectStatus.ACTIVE:
             outcome = quotas.backend.assign_monitor_seat(monitor)
             # The MonitorValidator checks if a seat assignment is availble.
             # This protects against a race condition
             if outcome != Outcome.ACCEPTED:
                 raise ParameterValidationError("Failed to enable monitor, please try again")
```

Comment:
if the update is for changing the status to active, this changes the monitor status without checking that we can actually assign a seat to that monitor, so maybe if the outcome isn't ACCEPTED we should call `monitor.update` again with the old status

Reply:
Hmm good point...  So ideally we should just update the slug separately ahead of time

Reply:
Yeah I think we definitely need to address this.